mat64/symmetric.go

```diff
@@ -177,8 +177,8 @@ func (s *SymDense) CopySym(a Symmetric) int {
 // the result in the receiver
 //  s = a + alpha * x * x'
 func (s *SymDense) SymRankOne(a Symmetric, alpha float64, x *Vector) {
-	n := s.mat.N
-	if x.Len() != n {
+	n := x.Len()
+	if a.Symmetric() != n {
 		panic(ErrShape)
 	}
 	var w SymDense
```

Comment:
The code below is wrong. We allocate a new w matrix and then test if it's zero.
````
func (s *SymDense) SymRankOne(a Symmetric, alpha float64, x *Vector) {
	n := x.Len()
	if a.Symmetric() != n {
		panic(ErrShape)
	}
	if s.isZero() {
		s.mat = blas64.Symmetric{
			N:      n,
			Stride: n,
			Uplo:   blas.Upper,
			Data:   use(s.mat.Data, n*n),
		}
	} else if n != s.mat.N {
		panic(ErrShape)
	}
	if s != a {
		s.CopySym(a)
	}
	blas64.Syr(alpha, x.mat, s.mat)
}
````